### PR TITLE
Adds experimental.OpenAIModel

### LINKED ISSUE #2636: Allow specifying the OpenAI model to use to generate commit messages
Now that we have `GitLens: Generate Commit Message`, I'd like to be able to configure the OpenAI model (GPT4 is really cool).

### PR DESCRIPTION
**Pull request by sadasant**
# Description

Adds a `gitlens.experimental.OpenAIModel` setting to specify the OpenAI model to use to generate commit messages when using the `GitLens: Generate Commit Message` command.

Fixes #2636

# Checklist

- [x] I have followed the guidelines in the Contributing document
- [x] My changes follow the coding style of this project
- [x] My changes build without any errors or warnings
- [x] My changes have been formatted and linted
- [x] My changes include any required corresponding changes to the documentation (including CHANGELOG.md and README.md)
- [x] My changes have been rebased and squashed to the minimal number (typically 1) of relevant commits
- [x] My changes have a descriptive commit message with a short title, including a `Fixes $XXX -` or `Closes #XXX -` prefix to auto-close the issue that your PR addresses

### CODE REVIEW

**Comment by sadasant** [comment]
This builds, linting passes, I was able to debug it and to see the model reflect in the debugger after specifying it in the configs (I tested with `gpt-4`). I also tested auto-completion.


https://user-images.githubusercontent.com/417016/232565357-94a4246c-7032-4f41-a009-8fb155ced2aa.mov

File: src/ai/openaiProvider.ts [hunk 1]
```diff
@@ -12,12 +12,15 @@ const maxCodeCharacters = 12000;
...
 	readonly id = 'openai';
 	readonly name = 'OpenAI';
+	private model: OpenAIChatCompletionModels = 'gpt-3.5-turbo';
 
 	constructor(private readonly container: Container) {}
 
 	dispose() {}
 
 	async generateCommitMessage(diff: string, options?: { context?: string }): Promise<string | undefined> {
+		this.model = configuration.get('experimental.openAIModel') || 'gpt-3.5-turbo';
```
Comment (d13) [inline src/ai/openaiProvider.ts:22]: Could pull this code into its own method (something like `ensureModel`) which is `(force = false): void`.  

```suggestion
		this.ensureModel();
```

**Comment by eamodio** [comment]
Thank you so much for your contribution!

Thank you!

File: src/ai/openaiProvider.ts (hunk 1 above)
Comment (sadasant) [reply src/ai/openaiProvider.ts]: Sorry, @d13 ! I focused on something else. I should have followed up sooner.

**Comment by sadasant** [comment]
Thank you all! I'll try to come back to this repo more often 🙇‍♂️

**Pull request merged**